htdocs/comm/propal/card.php

```diff
@@ -315,6 +315,9 @@
 		}
 
 		if (!$error) {
+			if (isset($object->duree_validite) && !empty($object->fin_validite)) {
+				$object->fin_validite = $datep + ($object->duree_validite * 24 * 3600);
```

Comment:
So you add code to avoid bad sync between duration of validity and end of validity. Why not.
But imagine, we initialy create a proposal the 1st of january with a validity duration of 10 days. So the end of validity is set during creation at 10th of january. Dolibarr is using this date for alerts.
Imagine the customer ask more delay. So the saler increase the end of validity date to 30th of january. But here, the code seems to overwrite this 30th of january to 10th of january. Am i wrong ?

Reply:
yes it is for a change of date. @eldy for you what would be the best solution?  To update the validity date.  take back the gap between the two dates?  or put options or offer to update the date?